### ISSUE TITLE
"Already defined error" with conditional importing

### ISSUE THREAD

**Issue by ngie-eign** [python/mypy#7557]
The following code triggers a false positive:
```
try:
    from unittest import mock  # python 3
except ImportError:
    import mock  # python 2
```
Example:
```
$ mypy ~/mypy_false_positive.py
mypy_false_positive.py:4: error: Name 'mock' already defined (by an import)
```

The false positive is tautilogically impossible, since `unittest.mock` failing to import would trigger the second statement on line 4.

This is a continuation of #1153.

I'm opening a new issue because #1153 has been closed for ~3 years now and I wanted to provide more up-to-date/salient details.

Additional details:
```
$ mypy --version
mypy 0.720
$ head -n 1 `which mypy`
#!/usr/bin/python3
$ python3 --version
Python 3.7.4
```

**Comment by ilevkivskyi** [comment]
That issue is still open (it was only closed for a day by mistake). I don't think we need a separate one for Python 2/3 use case. Especially that mypy supports another pattern:
```python
import sys
if sys.version_info >= (3, 0):
    from unittest import mock
else:
    import mock
```
